Assets/Scripts/ECS/Animal/Needs/MaxHungerSystem.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Unity.Entities;
+using Ecosystem.ECS.Death;
+
+namespace Ecosystem.ECS.Animal.Needs
+{
+    /// <summary>
+    /// System for clamping down hunger on an animal.
+    /// </summary>
+    public class MaxHungerSystem : SystemBase
+    {
+        private EndSimulationEntityCommandBufferSystem m_EndSimulationEcbSystem;
+
+        protected override void OnCreate()
+        {
+            m_EndSimulationEcbSystem = World
+                .GetOrCreateSystem<EndSimulationEntityCommandBufferSystem>();
+        }
+
+        protected override void OnUpdate()
+        {
+            var commandBuffer = m_EndSimulationEcbSystem.CreateCommandBuffer().ToConcurrent();
+
+            Entities.ForEach((Entity entity,int entityInQueryIndex,MaxHungerData maxHungerData,
+                ref HungerData hungerData) =>
+            {
+                    if (hungerData.Hunger >= maxHungerData.MaxHunger)
+                    {
+                        hungerData.Hunger = maxHungerData.MaxHunger;
+                    }
+
+                }).ScheduleParallel();
+
+            m_EndSimulationEcbSystem.AddJobHandleForProducer(Dependency);
```

Comment:
```suggestion
        protected override void OnUpdate()
        {
            Entities.ForEach((ref HungerData hungerData, in MaxHungerData maxHungerData) =>
            {
                    if (hungerData.Hunger >= maxHungerData.MaxHunger)
                    {
                        hungerData.Hunger = maxHungerData.MaxHunger;
                    }

                }).ScheduleParallel();
```
Behöver inte ha någon command buffer i det här systemet vilket ungefär halverar koden :)

Man behöver inte ha de parametrar som man inte använder i ForEach:en och varje komponent som man använder borde alltid ha 'ref' eller 'in' framför.

ref = read/write
in = read only